### PR TITLE
Error callback message

### PR DESCRIPTION
**Pull request by b-pmcg**
Allow for a error message parameter to return with the callback on Tx error.

### CODE REVIEW

File: src/eth/TransactionManager.js [hunk 1]
```diff
@@ -172,10 +172,16 @@ class Tracker {
     this._transactions[key].push(tx);
 
     for (let event of this.constructor.states) {
-      tx.on(event, () => this._listeners[key][event].forEach(cb => cb(tx)));
+      tx.on(event, () =>
+        this._listeners[key][event].forEach(
+          cb => (tx.error ? cb(tx, tx.error.message) : cb(tx))
```
Comment (b-pmcg) [inline src/eth/TransactionManager.js]: If there is an error property on the transaction, send the error message back with the callback function.

File: test/eth/TransactionManager.spec.js [hunk 2]
```diff
@@ -272,4 +272,34 @@ describe('lifecycle hooks', () => {
     expect(openTx.isFinalized()).toBe(true);
     expect(openHandlers.confirmed).toBeCalled();
   });
+
+  test('return error message with error callback', async () => {
+    const makeListener = () =>
+      jest.fn((tx, err) => {
```
Comment (b-pmcg) [inline test/eth/TransactionManager.spec.js]: The user can access the returned error message from their callback function argument.

**Comment by jparklev** [comment]
Lemme play with this locally a bit, be back soon

**Comment by jparklev** [comment]
Lemme try this locally a bit, will report back soon!

**Comment by jparklev** [comment]
Was having some trouble getting the `error` listener to trigger for reverted txs

Check this out to to see where I was experimenting with this (w/ the sdk locally linked) https://github.com/makerdao/governance-dashboard/blob/f39a38cf65bb241d00f273a77648d3e762db5ec1/src/reducers/proxy.js#L103

We could definitely produce a more minimal example case to confirm this behavior if it would seem to be useful. I'll note here if anything comes to mind as to why this might be

**Pull request merged**